src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -108,45 +115,49 @@ public void remove(Person toRemove) {
      * Sorts the Member's list based on a given predicate.
      */
     public void sortList(Predicate<String> predicate) {
+        requireNonNull(predicate);
+
         FXCollections.sort(internalList, new Comparator <Person> () {
             @Override
             public int compare(Person o1, Person o2) {
                 String sortCriteria = predicate.toString().toLowerCase();
-                String first = "";
-                String second = "";
-                switch (sortCriteria) {
-                    case "name":
-                        first = o1.getName().fullName;
-                        second = o2.getName().fullName;
-                        break;
-
-                    case "gender":
-                        first = o1.getGender().value;
-                        second = o2.getGender().value;
-                        break;
-
-                    case "major":
-                        first = o1.getMajor().value;
-                        second = o2.getMajor().value;
-                        break;
-
-                    case "yearofstudy":
-                        first = o1.getYearOfStudy().value;
-                        second = o2.getYearOfStudy().value;
-                        break;
-                    default:
+                Pair<String, String> test = getCriteria(sortCriteria, o1, o2);
+                int result = 1;
+                if (test == null) {
+                    result = test.getKey().compareTo(test.getValue());
                 }
-
-                int result = first.compareTo(second);
                 if (result != 0) {
                     return result;
-                } else {
-                    return o1.getName().fullName.compareTo(o2.getName().fullName);
                 }
+                return o1.getName().fullName.compareTo(o2.getName().fullName);
             }
         });
     }
 
+    /**
+     * [Add your comments]
+     * @param sortCriteria
+     * @param o1
+     * @param o2
+     * @return
+     */
+    public Pair<String, String> getCriteria(String sortCriteria, Person o1, Person o2) {
+        switch (sortCriteria) {
+        case SORT_BY_NAME:
+            return new Pair<>(o1.getName().fullName, o2.getName().fullName);
```

Comment:
one of the issues during PE dry run that people raised is that this sorting will have all the capital letters first. May consider whether to make the sort case insensitive?

Reply:
ok thank you!